packages/dos-donts-block/src/DosDontsBlock.tsx

```diff
@@ -1,95 +1,421 @@
+
+            setIsUploadLoading(false);
+        });
     };
 
-    const saveItem = (id: number, value: string, type: 'title' | 'body') => {
-        const updatedItems = localItems;
-        const existingItemIndex = localItems.findIndex((item) => item.id === id);
-        updatedItems[existingItemIndex] = { ...updatedItems[existingItemIndex], [type]: value };
+    const removeItemById = (itemId: string) => {
+        const newItems: Item[] = localItems.filter((item) => item.id !== itemId);
+        setAndSaveItems(newItems);
+    };
+
+    const onChangeItem = (
+        itemId: string,
+        value: string | number | undefined,
+        type: 'title' | 'body' | 'type' | 'imageId'
+    ) => {
+        const newItems: Item[] = localItems.map((item) => (item.id === itemId ? { ...item, [type]: value } : item));
+        setAndSaveItems(newItems);
+    };
 
+    const setAndSaveItems = (newItems: Item[]) => {
+        setLocalItems(newItems);
         setBlockSettings({
-            items: [...updatedItems],
+            ...blockSettings,
+            items: newItems,
         });
     };
+    const handleDragStart = (event: DragEndEvent) => {
+        const { active } = event;
+        setActiveId(active.id as string);
+        if (wrapperRef.current) {
+            setMinRowHeight(
+                Math.min(...Array.from(wrapperRef.current.children).map((x) => x.getBoundingClientRect().height))
+            );
+        }
+    };
 
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    useEffect(() => setItems(layout === DoDontLayout.Stacked ? columns * 2 : 2), [layout, columns]);
+    const handleDragEnd = (event: DragEndEvent) => {
+        const { active, over } = event;
+        if (over && active.id !== over.id) {
+            const oldIndex = localItems.findIndex((i) => i.id === active.id);
+            const newIndex = localItems.findIndex((i) => i.id === over.id);
+            const sortedItems = arrayMove(localItems, oldIndex, newIndex);
+            setLocalItems(sortedItems);
+            setBlockSettings({
+                ...blockSettings,
+                items: sortedItems,
+            });
+            setActiveId(undefined);
+        }
+    };
+
+    const getLinkedImageOfItem = (item: Item) => {
+        if (!itemImages) {
+            return;
+        }
+        return itemImages.find((asset) => asset.id === item.imageId)?.genericUrl;
+    };
+
+    const batchUpload = (type: DoDontType) => {
+        batchUploadType.current = type;
+        openFileDialog();
+    };
+
+    const openAssetChooser = (type: DoDontType) => {
+        batchUploadType.current = type;
+        appBridge.openAssetChooser(
+            (result) => {
+                batchAddItems(result);
+                appBridge.closeAssetChooser();
+            },
+            {
+                multiSelection: true,
+            }
+        );
+    };
+
+    const gridClassName = useMemo(() => {
```

Comment:
Same here with `useMemo`, is it really needed?

Reply:
Done, removed